src/bundle/Resources/public/js/scripts/admin.limitation.pick.js

```diff
@@ -1,19 +1,164 @@
+
+        input.value = locationsIdsWithoutRemoved.join(IDS_SEPARATOR);
+    };
+    const addLocationsTags = (limitationBtn, newlySelectedItems) => {
+        const tagsList = doc.querySelector(limitationBtn.dataset.selectedLocationListSelector);
+        const tagTemplate = limitationBtn.dataset.valueTemplate;
+        const fragment = doc.createDocumentFragment();
+
+        newlySelectedItems.forEach((location) => {
+            const locationId = location.id;
+            const container = doc.createElement('ul');
+            const renderedItem = tagTemplate.replace('{{ location_id }}', locationId);
+
+            container.insertAdjacentHTML('beforeend', renderedItem);
+
+            const listItemNode = container.querySelector('li');
+            const tagNode = listItemNode.querySelector(SELECTOR_EZ_TAG);
 
-        doc.querySelector(target.dataset.locationInputSelector).value = data.map((item) => item.id).join();
-        doc.querySelector(target.dataset.selectedLocationListSelector).innerHTML = selectedItems;
+            attachTagEventHandlers(limitationBtn, tagNode);
+            fragment.append(listItemNode);
+        });
+
+        tagsList.append(fragment);
+
+        setTagsBreadcrumbs(tagsList, newlySelectedItems);
+    };
+    const setTagsBreadcrumbs = (tagsList, newlySelectedItems) => {
+        const pathArraysWithoutRoot = newlySelectedItems.map(getLocationPathArray);
+
+        findLocationsByIdList(pathArraysWithoutRoot, (response) => {
+            const { operations } = response.BulkOperationResponse;
+
+            Object.entries(operations).forEach(([locationId, { content }]) => {
+                const viewData = JSON.parse(content);
+                const tag = tagsList.querySelector(`[data-location-id="${locationId}"]`);
+                const tagContent = tag.querySelector('.ez-tag__content');
+                const tagSpinner = tag.querySelector('.ez-tag__spinner');
+
+                tagContent.innerText = buildContentBreadcrumbs(viewData);
+
+                tagSpinner.hidden = true;
+                tagContent.hidden = false;
+            });
+        });
+    };
+    const getLocationPathArray = ({ pathString }) => {
+        const pathArray = pathStringToPathArray(pathString);
+        const pathArrayWithoutRoot = removeRootLocation(pathArray);
+
+        return pathArrayWithoutRoot;
+    };
+    const handleTagRemove = (limitationBtn, tag) => {
+        const removedLocationId = tag.dataset.locationId;
+        const locationInputSelector = limitationBtn.dataset.locationInputSelector;
+
+        removeLocationFromInput(locationInputSelector, removedLocationId);
+        tag.remove();
+    };
+    const attachTagEventHandlers = (limitationBtn, tag) => {
+        const removeTagBtn = tag.querySelector('.ez-tag__remove-btn');
+
+        removeTagBtn.addEventListener('click', handleTagRemove.bind(null, limitationBtn, tag), false);
```

Comment:
Why not: `() => handleTagRemove(limitationBtn, tag)`, it's shorter?

Reply:
Because I thought we prefer binds over arrow functions when it comes to `addEventListener`.

Reply:
If you just bind `this` then it's fine but when you're passing more params then I believe arrow functions are cleaner to read. In the background `bind` and arrow functions do the same.